### PR TITLE
Rename the GD7965 driver to UC81xx and add support for UC8176

### PR DESCRIPTION
**Pull request by andysan**
The GD7965 is really a driver for an UltraChip UC8179 controller. There are also EPD profiles in-tree that use the UC8176 controller which is mostly compatible (some registers have a slightly different layout).

This pull requests renames the driver to UC81xx to reflect the fact that it supports multiple UltraChip devices and it includes a number of changes to implement proper support for the UC8176. These changes should normally be included in two different pull requests, but I would like to avoid the risk of those PRs spanning two releasees since they would both affect the DT bindings.

The main user-facing change is the change of compatible strings. The gooddisplay,gd7965 has been replaced by chip-specific compatible strings (ultrachip,uc8176 and ultrachip,uv8179). There is no longer a generic driver string since the different chips need different quirks.

### CODE REVIEW

File: tests/drivers/build_all/sensor/spi.dtsi [hunk 1]
```diff
@@ -80,8 +80,20 @@ test_spi_lpd8806: lpd8806@9 {
 	spi-max-frequency = <0>;
 };
 
+test_spi_uc8176: uc8176@a {
+	compatible = "ultrachip,uc8176";
+	label = "UC8176";
```
Comment (mbolivar-nordic) [inline tests/drivers/build_all/sensor/spi.dtsi:85]: Remove this; we are phasing out use of label properties in upstream Zephyr, so let's not add any more, please.

File: drivers/display/uc81xx.c [hunk 2]
```diff
@@ -279,13 +274,9 @@ static int uc81xx_write(const struct device *dev, const uint16_t x, const uint16
...
 	}
 
-	if (config->override_cdi) {
-		/* Enable boarder output */
-		data->bdd_polarity &= ~UC81XX_CDI_BDZ;
-		if (uc81xx_write_cmd_uint8(dev, UC81XX_CMD_CDI,
-					   data->bdd_polarity)) {
-			return -EIO;
-		}
+	/* Enable boarder output */
```
Comment (mbolivar-nordic) [inline drivers/display/uc81xx.c:292]: Did you mean "border?" Here and above.

File: drivers/display/uc81xx.c [hunk 3]
```diff
@@ -279,13 +274,9 @@ static int uc81xx_write(const struct device *dev, const uint16_t x, const uint16
...
+	if (config->quirks->set_cdi(dev, true)) {
+		return -EIO;
```
Comment (mbolivar-nordic) [inline drivers/display/uc81xx.c:294]: Why swallow the error instead of propagating it up? Here and above.

File: drivers/display/uc81xx.c [hunk 4]
```diff
@@ -431,17 +421,8 @@ static int uc81xx_controller_init(const struct device *dev)
...
-		cdi[UC81XX_CDI_BDZ_DDX_IDX] = data->bdd_polarity;
-		cdi[UC81XX_CDI_CDI_IDX] = config->cdi;
-		LOG_HEXDUMP_DBG(cdi, sizeof(cdi), "CDI");
-		if (uc81xx_write_cmd(dev, UC81XX_CMD_CDI,
-				     cdi, sizeof(cdi))) {
-			return -EIO;
-		}
-	}
+	/* Set CDI and enable border output */
+	config->quirks->set_cdi(dev, true);
```
Comment (mbolivar-nordic) [inline drivers/display/uc81xx.c:432]: No error handling?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Overall looks quite nice. I think @jfischer-no should be the assignee however.

A few minor requests and questions from me.

File: tests/drivers/build_all/sensor/spi.dtsi (hunk 1 above)
Comment (andysan) [reply tests/drivers/build_all/sensor/spi.dtsi:85]: Done.

File: drivers/display/uc81xx.c (hunk 2 above)
Comment (andysan) [reply drivers/display/uc81xx.c:292]: Done.

**Review by jfischer-no — DISMISSED** [review]

File: boards/shields/waveshare_epaper/waveshare_epaper_gdew042t2.overlay [hunk 5]
```diff
@@ -7,14 +7,14 @@
...
 	};
 };
 
 &arduino_spi {
-	gd7965: gd7965@0 {
-		compatible = "gooddisplay,gd7965","gooddisplay,gdew042t2";
-    		label = "GD7965";
+	uc8176: uc8176@0 {
+		compatible =  "gooddisplay,gdew042t2", "ultrachip,uc81xx";
+		label = "UC8176";
```
[the remark this answers is not in the document]
Comment (andysan) [reply boards/shields/waveshare_epaper/waveshare_epaper_gdew042t2.overlay:17]: I just had a slightly annoying realisation. The LVGL integration still relies on labels in a few places so I think we'll need to keep these for now. :(

**Comment by andysan** [comment]
@mbolivar-nordic: I have rebased this change and fixed a couple of merge conflicts. Other than the merge conflicts, the change should be identical to when @jfischer-no approved it last week. Are we OK to merge this or do you want someone else to have a look?

**Review by mbolivar-nordic — APPROVED** [review]
Blindly reapproving to get you one of the +1s you need.

We need @jfischer-no to reapprove again as well due to our github repository configuration.

**Review by jfischer-no — APPROVED** [review]

**Comment by andysan** [comment]
Are we ok to merge this? I have some changes (fast partial refresh) that depend on this.

**Pull request merged**